File: random_number/android/src/main/kotlin/com/my/plugin/random_number/RandomNumberPlugin.kt
package com.my.plugin.random_number

import androidx.annotation.NonNull
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result

/** RandomNumberPlugin */
class RandomNumberPlugin : FlutterPlugin, MethodCallHandler {
  private val channelName = "com.my.plugin.random_number"
  private lateinit var channel: MethodChannel

  override fun onAttachedToEngine(
      @NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding
  ) {
    channel = MethodChannel(flutterPluginBinding.binaryMessenger, channelName)
    channel.setMethodCallHandler(this)
  }

  override fun onDetachedFromEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
    channel.setMethodCallHandler(null)
  }

  override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
    when (call.method) {
      "getPlatformVersion" -> result.success("Android ${android.os.Build.VERSION.RELEASE}")
      "luckyNumber" -> result.success(luckyNumber())
      else -> {
          result.notImplemented()
      }
    }
  }

  private fun luckyNumber(): Int {
      val lucky = (0..99).random()
      // channel.invokeMethod("luckyNumber", lucky)
      return lucky
  }
}

File: random_number/ios/Classes/RandomNumberPlugin.swift
import Flutter
import UIKit

public class RandomNumberPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "com.my.plugin.random_number", binaryMessenger: registrar.messenger())
    let instance = RandomNumberPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "getPlatformVersion":
      result("iOS " + UIDevice.current.systemVersion)
    case "luckyNumber":
      result(luckyNumber())
    default:
      result(FlutterMethodNotImplemented)
    }
  }

  private func luckyNumber() -> Int{
      let lucky = Int.random(in: 0 ... 100)
      // channel!.invokeMethod("luckyNumber", arguments: lucky)
      return lucky
  }
}
